pkg/network/status/status_test.go

```diff
@@ -55,6 +56,121 @@ var (
 	}
 )
 
+func TestProbeAllHosts(t *testing.T) {
+	hostA := "foo.bar.com"
+	hostB := "ksvc.test.dev"
+	hostBEnabled := int32(0)
+
+	ing := ingTemplate.DeepCopy()
+	ing.Spec.Rules[0].Hosts = append(ing.Spec.Rules[0].Hosts, hostB)
+	hash, err := ingress.InsertProbe(ing)
+	if err != nil {
+		t.Fatal("Failed to insert probe:", err)
+	}
+
+	// Dummy handler returning HTTP 500 (it should never be called during probing)
+	dummyRequests := make(chan *http.Request)
+	dummyHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		dummyRequests <- r
+		w.WriteHeader(500)
+	})
+
+	// Actual probe handler used in Activator and Queue-Proxy
+	probeHandler := network.NewProbeHandler(dummyHandler)
+
+	// Probes to hostA always succeed and probes to hostB only succeed if hostBEnabled is true
+	probeRequests := make(chan *http.Request)
+	finalHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		probeRequests <- r
+		if !strings.HasPrefix(r.Host, hostA) &&
+			(atomic.LoadInt32(&hostBEnabled) == 0 || !strings.HasPrefix(r.Host, hostB)) {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		r.Header.Set(network.HashHeaderName, hash)
+		probeHandler.ServeHTTP(w, r)
+	})
+
+	ts := httptest.NewServer(finalHandler)
+	defer ts.Close()
```

Comment:
t.Cleanup?

Reply:
For now, let's keep it consistent with the other tests.
We can revisit later and update the entire file to t.Cleanup if needed.